### ISSUE TITLE
Maven resource filtering - property replacement when using config server

### ISSUE THREAD

**Issue by tipsytopsy** [spring-cloud/spring-cloud-config#786]
Hello,

We've a spring boot application (maven) which has a bunch of environment based application.properties (-dev, -local, etc). We have a ton of property replacement using ${} delimiter and use maven-resources-plugin's delimiter config to not use default delimiter and use ${} instead. The other thing is that we've a lot of property values specified in pom.xml which have replacement references in application.properties.

This setup works fine and now we're trying to move to spring cloud config. There are two issues in cloud config because proper hooks are not available to help with property replacements.

1. The delimiter filtering is not possible when using cloud config. So we're forced to convert all ${} to @*@ in all properties.
2. The properties specified in pom.xml never make it to the properties retrieved from cloud config (obvious).

Can someone help with these issues/ concerns?

Thanks!

**Comment by spencergibb** [comment]
Number 2 won't happen. The config server itself isn't aware of its build tool. I'm still a bit unsure how maven's filtering affects using config server. Spring Cloud Config wasn't built to be used by a build tool.

**Comment by tipsytopsy** [comment]
Thanks @spencergibb .

Number 1 was little quirky.. Spring only required the maven properties to be delimited in @@. So I converted 4 of those properties (build.version, project.name, project.description and project.version) to @@ delimited and that was good. Previously the maven plugin used to do that at build time.

Number 2 - I agree that the config server external to an application has no knowledge of properties in the application. So I duplicated those properties to config server and all was good. I think there can be some hooks in the config-client where users can do some translations on client side. But cannot think of exactly what hook yet.

Thanks for your reply.. working with cloud config has been great so far!
